### DISCUSSION TITLE
Could someone explain this to me like I'm 10 years old (I'm 35 lol)

### DISCUSSION THREAD

**Discussion by alienkitty666** [icloud-photos-downloader/icloud_photos_downloader#628] [Q&A]
Hi everyone,

I'm generally able to figure out *anything* I need to make happen on a computer, no matter how complex, by just reading about it, but I can't get this to work for some reason. 

Is "icloudpd-1.13.0-windows-amd64.exe" the correct thing to download if I want to run icloudpd as an executable on Windows 10?

I've tried running it, and the window opens and asks me for my icloud username and when I hit enter it says something about a directory and immediately closes the window. I thought maybe I downloaded the wrong item from the release, so I tried the other two windows exe files and they close immediately after they open. 

What am I missing here? 

Thanks in advance, I've been trying to download my icloud photos various ways for an entire week now and am very excited to have found this tool! `

**Comment by AndreyNikiforov** [comment]
@alienkitty666 when you start cli app (command line interface) on windows, OS opens text windows. OS closes that window when an app completes, wiping out output of course. This is standard behavior for any cli (==non GUI) app.

On way to solve it:
- press WindowsKey-R; enter "cmd" in the prompt; click "Open" -- should open text windows that resembles terminal on other OSes.
- use "c:" (or other drive letter) to change current drive
- use "cd folder_name" to change current folder
- run icloudpd.exe there
- all output should be in that windows until you close the window

**Reply by alienkitty666** [reply]
It's not just that the window is closing when the app is complete though, it's closing before I have an opportunity to do anything. Surely it's not standard behavior for the window to close before I can actually run the downloader?

**Reply by alienkitty666** [reply]
I've just run it using the window opened from the command prompt as you outlined, and here is what happens- it prompts me for "iCloud username/email:" and when I enter that, it responds "--directory or --list-albums are required". So I tried entering the directory I want the photos downloaded to following my iCloud email and it responds with the same thing. I'm confused about what I'm supposed to be entering here, because it's just prompting me for my username/email and is different from what I thought I understood from the ReadMe. Is the format for what I need to enter different with the executable? 

If anyone can tell me the correct steps for what I need to enter here to run the program I would greatly appreciate it!

**Reply by AndreyNikiforov** [reply]
> I've just run it using the window opened from the command prompt as you outlined, and here is what happens- it prompts me for "iCloud username/email:" and when I enter that, it responds "--directory or --list-albums are required". So I tried entering the directory I want the photos downloaded to following my iCloud email and it responds with the same thing. I'm confused about what I'm supposed to be entering here, because it's just prompting me for my username/email and is different from what I thought I understood from the ReadMe. Is the format for what I need to enter different with the executable?
> 
> If anyone can tell me the correct steps for what I need to enter here to run the program I would greatly appreciate it!

Command Line Interface programs assume parameters passed at start, e.g.: "icloudpd.exe --directory where_i_want_to_save_photos --username my@email.address". You can see all supported parameters if you enter "icloudpd.exe --help"

**Reply by alienkitty666** [reply]
I really appreciate your help. I don't know what you mean when you say it "assumes parameters passed at start". I understand what all the parameters are, because I read the ReadMe. But no matter how I type them in, I'm getting the same response from the program. When it prompts me "iCloud username/email:" what ---exactly--- am I supposed to type to make this program work? All I am interested in specifying is the folder for it to download to, none of the other options. 

I also did try entering "icloudpd.exe --help" and it just kicks back the same thing as everything else I put in- "--directory or --list-albums are required" 

I feel like I'm missing something here, which is why I asked for the steps to be explained like I'm 10 years old haha. I'm not a computer illiterate person so I don't understand why I can't get this to work.
